internal/session/session.go

```diff
@@ -153,8 +158,14 @@ func (s *Session) serve(ctx context.Context, cmdCh <-chan command) error {
 			}
 
 		default:
-			for res := range s.handleOther(withStartTime(ctx, time.Now()), tag, cmd) {
+			responseCh := s.handleOther(withStartTime(ctx, time.Now()), tag, cmd)
+			for res := range responseCh {
 				if err := res.Send(s); err != nil {
+					// Consume all remaining channel response since the connection is no longer available.
+					// Failing to do so can cause a deadlock in the program as `s.handleOther` never finishes
+					// executing and can hold onto a number of locks indefinitely.
+					for _ = range responseCh {
```

Comment:
could be good to add a TODO here to track it on Jira and make sure we can't find a better way to handle the error

Reply:
I'll create a ticket in JIRA just in case.